tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -925,6 +925,65 @@ var _ = Describe("odo devfile push command tests", func() {
 		Context("odo push (without -f) is executed", func() {
 			ensureFilesSyncedTest(commonVar.Project, false)
 		})
+
+		When("node-js application is created and pushed", func() {
+			var output string
+			BeforeEach(func() {
+				helper.Cmd("odo", "create", "--project", commonVar.Project, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-MR-CL-CR.yaml")).ShouldPass()
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				output = helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass().Out()
+			})
+
+			It("should push the component successfully", func() {
+				Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+			})
+			It("should check for cpuLimit", func() {
+				podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+				output = helper.Cmd("oc", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.limits.cpu}'").ShouldPass().Out()
+				Expect(output).To(ContainSubstring("1"))
+			})
+			It("should check for cpuRequests", func() {
+				podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+				output = helper.Cmd("oc", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.requests.cpu}'").ShouldPass().Out()
+				Expect(output).To(ContainSubstring("200m"))
+			})
+			It("should check for memoryRequests", func() {
+				podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+				output = helper.Cmd("oc", "get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.requests.memory}'").ShouldPass().Out()
+				Expect(output).To(ContainSubstring("512Mi"))
+			})
+			When("Update the devfile.yaml, do odo push", func() {
+
+				BeforeEach(func() {
+					helper.ReplaceString("devfile.yaml", "cpuLimit: \"1\"", "cpuLimit: 700m")
```

Comment:
```suggestion
					helper.ReplaceString("devfile.yaml", "cpuLimit: 1", "cpuLimit: 700m")
```
I think the above should also work.

Reply:
This will not work as the value needs to be string. I have tried the changes you shared above while writing test.